ports/servoshell/minibrowser.rs

```diff
@@ -56,18 +64,62 @@ impl Minibrowser {
             .egui_ctx
             .set_pixels_per_point(window.hidpi_factor().get());
 
+        let widget_surface_fbo = match webrender_surfman.context_surface_info() {
+            Ok(Some(info)) => info.framebuffer_object,
+            Ok(None) => panic!("Failed to get widget surface info from surfman!"),
+            Err(error) => panic!(
+                "Failed to get widget surface info from surfman! {:?}",
+                error
+            ),
+        };
+
         Self {
             context,
             event_queue: RefCell::new(vec![]),
             toolbar_height: Default::default(),
+            widget_surface_fbo,
             last_update: Instant::now(),
+            last_mouse_position: None,
             location: RefCell::new(initial_url.to_string()),
             location_dirty: false.into(),
         }
     }
 
+    /// Preprocess the given [winit::event::WindowEvent], returning unconsumed for mouse events in
+    /// the Servo browser rect.
+    pub fn on_event(&mut self, event: &winit::event::WindowEvent<'_>) -> EventResponse {
+        let mut result = self.context.on_event(event);
+        result.consumed &= match event {
+            winit::event::WindowEvent::CursorMoved { position, .. } => {
+                let scale = Scale::<_, DeviceIndependentPixel, _>::new(
+                    self.context.egui_ctx.pixels_per_point(),
+                );
+                self.last_mouse_position =
+                    Some(winit_position_to_euclid_point(*position).to_f32() / scale);
+                self.last_mouse_position
+                    .map_or(false, |p| self.is_in_browser_rect(p))
+            },
+            winit::event::WindowEvent::MouseWheel { .. } |
+            winit::event::WindowEvent::MouseInput { .. } => self
+                .last_mouse_position
+                .map_or(false, |p| self.is_in_browser_rect(p)),
+            _ => true,
+        };
+        result
+    }
```

Comment:
Hrm. Out of curiosity, how does this relate to offscreen rendering?

Reply:
we use offscreen rendering in servoshell by blitting the webview to an egui [CentralPanel](https://docs.rs/egui/0.22.0/egui/containers/panel/struct.CentralPanel.html) when the minibrowser is enabled. but adding the CentralPanel makes egui report events that land in the webview area as consumed. clarified in f299776edded5fc6abd5cd8b1347b92942d58a4d